upgrades/steps_27.go

```diff
@@ -64,3 +81,85 @@ func stateStepsFor27() []Step {
 		},
 	}
 }
+
+// stepsFor27 returns upgrade steps for Juju 2.7.
+func stepsFor27() []Step {
+	return []Step{
+		&upgradeStep{
+			description: "change owner of unit and machine logs to adm",
+			targets:     []Target{AllMachines},
+			run:         resetLogPermissions,
+		},
+	}
+}
+
+func setJujuFolderPermissionsToAdm(dir string) error {
+	wantedOwner, wantedGroup := systemd.SyslogUserGroup()
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		filePath := dir + string(os.PathSeparator) + info.Name()
+		if err := auditlog.SetOwnerGroupLogPermissions(filePath, wantedOwner, wantedGroup, 0640); err != nil {
```

Comment:
I think we should have two steps here for the ownership and the permissions, because most cases, other call sites don't care about the permission.